tools/build_api.py

```diff
@@ -234,17 +234,21 @@ def find_valid_toolchain(target, toolchain):
     last_error = None
     for index, toolchain_name in enumerate(toolchain_names):
         internal_tc_name = get_toolchain_name(target, toolchain_name)
-        if toolchain == "ARM" and toolchain_name == "ARMC5" and index != 0:
+        if toolchain_name == "ARMC5":
             end_warnings.append(ARMC5_MIGRATION_WARNING)
+            if target.default_toolchain == "uARM":
+                end_warnings.append(UARM_DEFAULT_TOOLCHAIN_WARNING)
         if not TOOLCHAIN_CLASSES[internal_tc_name].check_executable():
             search_path = TOOLCHAIN_PATHS[internal_tc_name] or "No path set"
             last_error = (
                 "Could not find executable for {}.\n"
                 "Currently set search path: {}"
             ).format(toolchain_name, search_path)
         else:
-            if toolchain_name == "uARM" or target.default_toolchain == "uARM":
-                 end_warnings.append(UARM_TOOLCHAIN_WARNING)
+            if toolchain_name == "uARM":
+                end_warnings.append(UARM_TOOLCHAIN_WARNING)
+            elif toolchain_name == "ARMC6" and target.default_toolchain == "uARM":                
```

Comment:
Why are you checking for `target.default_toolchain`?
Also isn't there an opportunity do so something like:
```
if (
    toolchain_name in ["uARM", "ARMC5", "ARMC6"]
    and target.default_toolchain == "uARM"
):
    end_warnings.append(UARM_TOOLCHAIN_WARNING)
```
(check if the syntax is correct)

Reply:
For me, this logic has become too complex to maintain. Please can you take the opportunity to refactor this into a sequence of 'if' statements at the same level, rather than the current complex arrangement.

Reply:
> Why are you checking for target.default_toolchain?
> Also isn't there an opportunity do so something like:
> if (
>     toolchain_name in ["uARM", "ARMC5", "ARMC6"]
>     and target.default_toolchain == "uARM"
> ):
>     end_warnings.append(UARM_TOOLCHAIN_WARNING)
> 
> (check if the syntax is correct)

The reason for checking target.default_toolchain is, if the target tries to build with ARM(ARMC5 or ARMC6) toolchain but the target.default_toolchain is configured to uARM then internally build tool use ARM with Small library.

So code snapshot will not work for all the target, such as LPC11C24 has uARM in the  supported_toolchain but there is no target.default_toolchain.

Reply:
Thanks.
Please refactor the code to avoid repetitions and add comment where an explanation is needed.